CDAlertView/Classes/CDAlertView.swift

```diff
@@ -507,14 +526,18 @@ open class CDAlertView: UIView {
         headerView = CDAlertHeaderView(type: type, isIconFilled: isHeaderIconFilled)
         headerView.backgroundColor = UIColor.clear
         headerView.hasShadow = hasShadow
+        headerView.hasRoundCorners = hasRoundCorners
         headerView.alertBackgroundColor = alertBackgroundColor
         headerView.circleFillColor = circleFillColor
+        if circleFillColor == .clear {
```

Comment:
This comes with the bug in the header where we also put a shadow on it. So whenever you set it to clear and remove circle header, the shadow will stay. I was thinking to add this functionality by using `CDAlertViewType` instead of doing like this. Therefore, I would prefer the other approach where type defines the image style.

Also, when the user wants to use an icon (via `.custom(image:)` type) without a color, it will fail. The header will shrink, the image won't be displayed and there will be a half circle as a header.

Reply:
I never thought of a type something like "noHeaderImage" which as you said is probably a better way to go.